cpp/src/io/utilities/row_selection.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <io/utilities/row_selection.hpp>
+
+#include <cudf/utilities/error.hpp>
+
+#include <algorithm>
+#include <limits>
+
+namespace cudf::io::detail {
+
+std::pair<uint64_t, size_type> skip_rows_num_rows_from_options(
+  uint64_t skip_rows_opt, std::optional<size_type> const& num_rows_opt, uint64_t num_source_rows)
+{
+  auto const rows_to_skip = std::min(skip_rows_opt, num_source_rows);
+  if (not num_rows_opt.has_value()) {
+    CUDF_EXPECTS(num_source_rows - rows_to_skip <= std::numeric_limits<size_type>::max(),
+                 "ORC reader can't read all rows from its input(s)");
```

Comment:
Clarifying the wording, the problem isn't that the user is necessarily trying to read all the lines, it's that the number of lines requested exceeds the cudf size_type limit
```suggestion
                 "The requested number of rows to read exceeds the largest cudf column size");
```